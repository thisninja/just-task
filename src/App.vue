<template>
  <div class="app__wrapper">
    <md-toolbar v-if="auth" class="md-dense">
      <h3 class="md-title toolbar">Welcome, {{ email }}</h3>
      <md-button class="md-primary" @click="onLogout">
        {{ LOGOUT_TITLE }}
      </md-button>
    </md-toolbar>
    <div class="app__container">
        <transition name="slide" mode="out-in">
          <router-view></router-view>
        </transition>
    </div>
  </div>
</template>

<script>
const LOGOUT_TITLE = 'LOGOUT';

import { mapGetters } from 'vuex';

export default {
  name: 'App',
  data () {
    return {
      LOGOUT_TITLE,
    }
  },
  computed: {
    ...mapGetters({
      email: 'email',
      auth: 'isAuthenticated',
    }),
  },
  methods: {
    onLogout() {
      // @TODO
    }
  },
}
</script>

<style lang="scss">
@import '@/assets/scss/_variables.scss';

.app {
  &__wrapper {
    font-family: 'Avenir', Helvetica, Arial, sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;

    .toolbar.md-title  {
      flex: 1;
      font-size: 14px;
      color: $material-grey-200;
    }
  }

  &__container {
    display: flex;
    align-items: center;
    justify-content: center;
    flex-direction: column;
    max-width: 500px;
    margin: 100px auto;
    text-align: center;
  }
}

.layout__icon {
  width: 58px;
  height: 58px;
}

.layout__link {
  margin-top: 10px;
  color: $material-grey-200;
  &:hover {
    color: $material-grey-200;
  }

  span {
    color: $material-blue-200;
  }
}

.auth-form {
  margin-bottom: 10px;
}

.full-width {
  width: 100%;
}

.slide-enter-active {
    animation: slide-in 200ms ease-out forwards;
}

.slide-leave-active {
    animation: slide-out 200ms ease-out forwards;
}

@keyframes slide-in {
    from {
        transform: translateY(-30px);
        opacity: 0;
    }
    to {
        transform: translateY(0);
        opacity: 1;
    }
}

@keyframes slide-out {
    from {
        transform: translateY(0);
        opacity: 1;
    }
    to {
        transform: translateY(-30px);
        opacity: 0;
    }
}
</style>
