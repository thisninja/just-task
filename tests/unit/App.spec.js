import Vue from 'vue';
import Vuex from 'vuex';
import VueMaterial from 'vue-material';
import VueRouter from 'vue-router';
import { shallowMount, createLocalVue } from '@vue/test-utils';
import App from '@/App.vue';

const EventBus = new Vue();

const GlobalPlugins = {
  install(v) {
    // Event bus
    v.prototype.$bus = EventBus;
  },
};

const localVue = createLocalVue();

localVue.use(GlobalPlugins);
localVue.use(VueRouter);
localVue.use(Vuex);
localVue.use(VueMaterial);

describe('App.vue', () => {
  let wrapper;
  let store;
  let actions;
  let mutations;
  let getters;

  beforeEach(() => {
    getters = getters || {
      email: () => '',
      isAuthenticated: () => null,
    };

    actions = actions || {
      logout: jest.fn(),
      keepSessionPersistent: jest.fn(),
    };

    mutations = mutations || {
      clearAuthData: jest.fn(),
      authUser: jest.fn(),
    };

    store = new Vuex.Store({
      getters,
      actions,
      mutations,
    });

    wrapper = shallowMount(App, {
      store,
      localVue,
    });
  });

  afterEach(() => {
    getters = null;
    actions = null;
    mutations = null;
  });

  it('should match snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe('computed', () => {
    beforeAll(() => {
      getters = {
        email: () => 'test@test.com',
        isAuthenticated: () => true,
      };
    });

    it('should match snapshot when auth true', () => {
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('created hook', () => {
    it('EventBus', () => {
      const spy = jest.spyOn(wrapper.vm.$bus, '$emit');
      const mockMsg = 'foo';
      wrapper.vm.$bus.$emit('token-validation:failed', mockMsg);

      expect(spy).toHaveBeenCalled();

      // expect(wrapper.vm.errorMessage).toEqual(mockMsg);
      // expect(wrapper.vm.showSnackbar).toEqual(true);
    });

    it('keepSessionPersistent', () => {
      expect(actions.keepSessionPersistent).toHaveBeenCalled();
    });
  });
});