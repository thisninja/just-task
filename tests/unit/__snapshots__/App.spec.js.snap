// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App.vue computed should match snapshot when auth true 1`] = `
<div class="app__wrapper">
  <md-toolbar-stub mdelevation="4" class="md-dense">
    <h3 class="md-title toolbar">Welcome, test@test.com</h3>
    <md-button-stub mdripple="true" type="button" class="md-primary">
      LOGOUT
    </md-button-stub>
  </md-toolbar-stub>
  <div class="app__container">
    <router-view-stub name="slide" mode="out-in"></router-view-stub>
  </div>
  <md-snackbar-stub mdpersistent="true" mdduration="4000" mdposition="center">

  </md-snackbar-stub>
</div>
`;

exports[`App.vue should match snapshot 1`] = `
<div class="app__wrapper">
  <!---->
  <div class="app__container">
    <router-view-stub name="slide" mode="out-in"></router-view-stub>
  </div>
  <md-snackbar-stub mdpersistent="true" mdduration="4000" mdposition="center">

  </md-snackbar-stub>
</div>
`;
